test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/Roundtrip/JsonReaderWriterInjectionTests.cs

```diff
@@ -37,6 +38,59 @@ public void UseDefaultJsonReader()
             RunReaderTest(null, "{\"@odata.context\":\"http://test/$metadata#People/$entity\",\"PersonId\":999,\"Name\":\"Jack\"}");
         }
 
+        [Fact]
+        public void UseComplexJsonReaderNestedSelect()
+        {
+            // $select =Address/SubAddress/SubRoad OR
+            //$select =Address($select =SubAddress($select=SubRoad))
+
+            var resDict =  RunComplexReaderTest(null, "{\"@odata.context\":\"http://test/$metadata#People(Address/SubAddress/SubRoad)/$entity\","+
+            "\"Address\":{" +
+                "\"SubAddress\":{\"SubRoad\":\"Redmond\"}}}");
+
+            Assert.Equal(resDict["DefaultNs.SubAddress"].ToList()[0].Value, "Redmond");
+        }
+        
+        [Fact]
+        public void UseComplexJsonReaderExpandAndSelect()
+        {
+            // $Expand =PersonCity($select =Zipcode),Address/City
+
+            var resDict = RunComplexReaderTest(null, "{\"@odata.context\":\"http://test/$metadata#People(PersonCity(ZipCode),Address/City())/$entity\"," +
+            "\"PersonCity\":{\"ZipCode\":98052},"+

```

Comment:
>98052 [](start = 41, length = 5)

use different ZipCode and test that can distinguish PersionCity from City cleary

Reply:
Resolved